src/components/data/DataTable.js

```diff
@@ -308,6 +308,226 @@ const DataTable = props => {
+                        { label: 'Delete Column', disabled: !!noEdit, tooltip: noEdit || '', onClick: () => setTimeout(() => setDeletingColumn(attributeName), 0) },
+                        { label: 'Clear Column', disabled: !!noEdit, tooltip: noEdit || '', onClick: () => setTimeout(() => setClearingColumn(attributeName), 0) }
+                      ] : [],
+                      extraColumnActions ? extraColumnActions(attributeName) : [])
+                  }, [
+                    h(HeaderCell, [
+                      !!columnNamespace && span({ style: { fontStyle: 'italic', color: colors.dark(0.75), paddingRight: '0.2rem' } }, [columnNamespace]),
+                      columnName
+                    ])
+                  ])
+                ]),
+                cellRenderer: ({ rowIndex }) => {
+                  const { attributes: { [attributeName]: dataInfo }, name: entityName } = entities[rowIndex]
+                  const dataCell = renderDataCell(dataInfo, workspace)
+                  const divider = div({ style: { flexGrow: 1 } })
+                  const copyButton = h(ClipboardButton, {
+                    'aria-label': `Copy attribute ${attributeName} of ${entityType} ${entityName} to clipboard`,
+                    className: 'cell-hover-only',
+                    style: { marginLeft: '1rem' },
+                    text: entityAttributeText(dataInfo)
+                  })
+                  const editLink = editable && h(EditDataLink, {
+                    'aria-label': `Edit attribute ${attributeName} of ${entityType} ${entityName}`,
+                    'aria-haspopup': 'dialog',
+                    'aria-expanded': !!updatingEntity,
+                    onClick: () => setUpdatingEntity({ entityName, attributeName, attributeValue: dataInfo })
+                  })
+
+                  if (!!dataInfo && _.isArray(dataInfo.items)) {
+                    const isPlural = dataInfo.items.length !== 1
+                    const label = dataInfo?.itemsType === 'EntityReference' ?
+                      isPlural ? 'entities' : 'entity' :
+                      isPlural ? 'items' : 'item'
+                    const itemsLink = h(Link, {
+                      style: { display: 'inline', whiteSpace: 'nowrap', marginLeft: '1rem' },
+                      onClick: () => setViewData(dataInfo)
+                    }, ` (${dataInfo.items.length} ${label})`)
+                    return h(Fragment, [dataCell, divider, copyButton, editLink, itemsLink])
+                  } else {
+                    return h(Fragment, [dataCell, divider, copyButton, editLink])
+                  }
+                }
+              }
+            }, visibleColumns)]
+
+            const defaultColumns = [
+              ...(dataProvider.features.supportsRowSelection ? defaultColumnsWithSelectRow : defaultColumnsWithoutSelectRow),
+            ]
```

Comment:
you could write this as:
```javascript
            const defaultColumns =
               [...(dataProvider.features.supportsRowSelection ? selectRowColumn : []), ...defaultColumnsWithoutSelectRow]
```
and delete the `defaultColumnsWithSelectRow` const entirely, which avoids all the duplicated code